From 6fa2ab23adf6d7de781e906d04a0224410262d9b Mon Sep 17 00:00:00 2001
Message-Id: <6fa2ab23adf6d7de781e906d04a0224410262d9b.1425610844.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Mon, 23 Feb 2015 06:26:01 -0500
Subject: [PATCH 01/27] FOR_UPSTREAM [VPG]: drm/i915: Re-adjusting rc6
 promotional timer for chv

After feedback from the hardware team we are changing the RC6 promotional
timer to increase the power saving without changing performance.

Issue: GMINL-6362
Change-Id: I74856e616bbf6ec6ee673db141fe15000fec73f1
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h |    1 +
 drivers/gpu/drm/i915/intel_pm.c |   13 +++++++++----
 2 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 65701ea..514a2e5 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -6002,6 +6002,7 @@ enum punit_power_well {
 #define GEN6_RC1e_THRESHOLD			0xA0B4
 #define GEN6_RC6_THRESHOLD			0xA0B8
 #define GEN6_RC6_RENDER_PROMOTION_TIMER_TO	0x0557
+#define CHV_RC6_RENDER_PROMOTION_TIMER_TO	0x0186
 #define GEN6_RC6_MEDIA_PROMOTION_TIMER_TO	0x00C3
 #define GEN6_RC6p_THRESHOLD			0xA0BC
 #define VLV_RCEDATA				0xA0BC
diff --git a/drivers/gpu/drm/i915/intel_pm.c b/drivers/gpu/drm/i915/intel_pm.c
index c2c99f9..a0e30a5 100644
--- a/drivers/gpu/drm/i915/intel_pm.c
+++ b/drivers/gpu/drm/i915/intel_pm.c
@@ -3864,9 +3864,14 @@ void vlv_modify_rc6_promotion_timer(struct drm_i915_private *dev_priv,
 	if (media_active)
 		I915_WRITE(GEN6_RC6_THRESHOLD,
 				GEN6_RC6_MEDIA_PROMOTION_TIMER_TO);
-	else
-		I915_WRITE(GEN6_RC6_THRESHOLD,
-				GEN6_RC6_RENDER_PROMOTION_TIMER_TO);
+	else {
+		if (IS_CHERRYVIEW(dev_priv->dev))
+			I915_WRITE(GEN6_RC6_THRESHOLD,
+					CHV_RC6_RENDER_PROMOTION_TIMER_TO);
+		else
+			I915_WRITE(GEN6_RC6_THRESHOLD,
+					GEN6_RC6_RENDER_PROMOTION_TIMER_TO);
+	}
 }
 
 static void vlv_media_timeout_work_func(struct work_struct *work)
@@ -4683,7 +4688,7 @@ static void cherryview_enable_rps(struct drm_device *dev)
 		I915_WRITE(RING_MAX_IDLE(ring->mmio_base), 10);
 	I915_WRITE(GEN6_RC_SLEEP, 0);
 
-	I915_WRITE(GEN6_RC6_THRESHOLD, 0x557);
+	I915_WRITE(GEN6_RC6_THRESHOLD, CHV_RC6_RENDER_PROMOTION_TIMER_TO);
 
 	/* allows RC6 residency counter to work */
 	I915_WRITE(VLV_COUNTER_CONTROL,
-- 
1.7.9.5

